packages/unchained-client/src/evm/ethereum/parser/thor.ts

```diff
@@ -18,26 +23,50 @@ export interface ParserArgs {
   rpcUrl: string
 }
 
+interface SupportedFunctions {
+  depositSigHash: string
+  transferOutSigHash: string
+}
+
 export class Parser implements SubParser<Tx> {
   readonly routerContract: string
-  readonly abiInterface = new ethers.utils.Interface(THOR_ABI)
-
-  readonly supportedFunctions = {
-    depositSigHash: this.abiInterface.getSighash('deposit'),
-    transferOutSigHash: this.abiInterface.getSighash('transferOut'),
-  }
+  readonly abiInterface: ethers.utils.Interface
+  readonly supportedFunctions: SupportedFunctions
 
   constructor(args: ParserArgs) {
+    this.abiInterface = (() => {
+      switch (args.chainId) {
+        case ethChainId:
+        case 'eip155:3':
+          return new ethers.utils.Interface(THOR_ETHEREUM_ABI)
+        case avalancheChainId:
+          return new ethers.utils.Interface(THOR_AVALANCHE_ABI)
+        default:
+          throw new Error(
+            `chainId is not supported. (supported chainIds: ${ethChainId}, ${avalancheChainId}, eip155:3)`,
+          )
+      }
+    })()
+
+    this.supportedFunctions = {
+      depositSigHash: this.abiInterface.getSighash('deposit'),
+      transferOutSigHash: this.abiInterface.getSighash('transferOut'),
+    }
+
     // TODO: Router contract can change, use /inbound_addresses endpoint to determine current router contract.
     // We will also need to know all past router contract addresses if we intend on using receive address as the means for detection
     this.routerContract = (() => {
       switch (args.chainId) {
-        case 'eip155:1':
-          return THOR_ROUTER_CONTRACT_MAINNET
+        case ethChainId:
+          return THOR_ROUTER_CONTRACT_ETH_MAINNET
+        case avalancheChainId:
+          return THOR_ROUTER_CONTRACT_AVAX_MAINNET
         case 'eip155:3':
           return THOR_ROUTER_CONTRACT_ROPSTEN
```

Comment:
we should probably just remove this and all references to ropsten to avoid confusion - we don't have wallet/caip/unchained support for any of this

Reply:
https://github.com/shapeshift/lib/pull/1091/commits/84b5cacb85641b12954974e79bcb1afe41a62bdb